src/coreclr/nativeaot/Runtime/CMakeLists.txt

```diff
@@ -203,11 +203,11 @@ if (CLR_CMAKE_TARGET_ARCH_AMD64 AND CLR_CMAKE_TARGET_WIN32)
   )
 endif (CLR_CMAKE_TARGET_ARCH_AMD64 AND CLR_CMAKE_TARGET_WIN32)
 
-if (NOT (CLR_CMAKE_TARGET_ARCH_AMD64 OR CLR_CMAKE_TARGET_ARCH_ARM64))
+if (CLR_CMAKE_TARGET_ARCH_ARM)
   list(APPEND RUNTIME_SOURCES_ARCH_ASM
     ${ARCH_SOURCES_DIR}/Interlocked.${ASM_SUFFIX}
   )
-endif (NOT (CLR_CMAKE_TARGET_ARCH_AMD64 OR CLR_CMAKE_TARGET_ARCH_ARM64))
+endif (CLR_CMAKE_TARGET_ARCH_ARM)
```

Comment:
What about X86?

Reply:
What about it? The interlocked assembly code for it is gone.

Reply:
Ah right I forgot that filip moved the 64bit overload to C++.